### PR TITLE
Add option to UTF8 Encode All Objects

### PR DESCRIPTION
**Pull request by OliverVahey**
Added UTF8 support for importing and exporting all objects from Access

### CODE REVIEW

File: Modules/modExtendSaveAsText.ACM [hunk 1]
```diff
@@ -154,65 +154,81 @@ End Sub
...
+        If ConvFileUTF8_to_UCS2(strFilename, strFilename & ".tmp") Then
+          strFilename = strFilename & ".tmp"
+        Else
+          strErr = "Unable to Convert '" & strFilename & "' from UTF-8 encoding to UCS2 so it has not imported. Please review after import."
+        End If
+        
+      
+      Case Else
+        If ConvFileUTF8_to_ANSI(strFilename, strFilename & ".tmp") Then
+          strFilename = strFilename & ".tmp"
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: I would give this variable a different name, maybe `strTemp` or even `str`. I found it confusing that `strFilename` is either the original file name, or the temp file path.

File: Modules/modFileEncoding.ACM [hunk 2]
```diff
@@ -521,4 +521,15 @@ Private Function bit16ShiftR(i As Long, s As Integer) As Long
   bit16ShiftR = Fix(bit16ShiftR / (2 ^ s))
   
   bit16ShiftR = bit16ShiftR And 65535 '&HFFFF ' ensure we only have least significant 16 bits
+End Function
+
+Function ConvFileUTF8_to_ANSI(strFilename As String, Optional strNewFilename As String) As Boolean
```
Comment (gilbertbw) [inline Modules/modFileEncoding.ACM:526]: Should this be private?

File: Forms/Access SVN - Options.ACF [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Version =20
```
Comment (gilbertbw) [inline Forms/Access SVN - Options.ACF]: You have reencoded this file!

File: Forms/Access SVN - Options.ACF [hunk 4]
```diff
@@ -2419,14 +2419,14 @@ Begin Form
                     OverlapFlags =93
                     Left =375
                     Top =7260
-                    Width =2025
-                    Height =210
+                    Width =2040
+                    Height =240
                     Name ="lblUTF8"
-                    Caption ="Convert Encoding to UTF-8"
+                    Caption ="Convert to UTF-8 (Old)"
```
Comment (gilbertbw) [inline Forms/Access SVN - Options.ACF]: Maybe "Deprecated" rather than "Old"?

File: Modules/modExtendSaveAsText.ACM [hunk 5]
```diff
@@ -133,21 +139,50 @@ End_Processing:
...
         If Not ConvFileUCS2_to_UTF8(strFilename) Then 'convert file to UTF8
           MsgBox "Unable to convert '" & objName & "' to UTF-8 encoding. Beware when committing."
         End If
+      ElseIf (enmEncode = asEncode_All_to_UTF8) Then
+        If Not ConvFile_to_UTF8(strFilename) Then
+          MsgBox "Unable to convert '" & objName & "' to UTF-8 encoding. Beware when committing."
+        End If
       End If
     Case asObjectType.acModule
       Application.SaveAsText objType, objName, strFilename ' no conversion needed - modules are output as ANSI
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: this comment is no longer correct!

File: Modules/modExtendSaveAsText.ACM [hunk 6]
```diff
@@ -133,21 +139,50 @@ End_Processing:
...
+        End If
+      End If
     Case asObjectType.acReference
       SaveReferencesAsText strFilename
+      If (enmEncode = asEncode_All_to_UTF8) Then
+        If Not ConvFile_to_UTF8(strFilename) Then
+          MsgBox "Unable to convert '" & objName & "' to UTF-8 encoding. Beware when committing."
+        End If
+      End If
   End Select
```
Comment (gilbertbw) [inline Modules/modExtendSaveAsText.ACM]: Do you think these should be refactored?

File: Forms/Access SVN - Options.ACF [hunk 7]
```diff
@@ -2740,6 +2793,50 @@ Private Sub cmdUTF8Encoding_Click()
...
+          End If
+        End If
+      Next
+    End If
+  Next
+  
+  SetConfigValue "UTF-8 Encode All", True, Me.txtOutputLocation.Value
+    
+  Me.cmdRun.SetFocus
+  Me.cmdUTF8FullEncode.Enabled = False
```
Comment (gilbertbw) [inline Forms/Access SVN - Options.ACF:2836]: The 'convert encoding to UTF-8' button should also be disabled, as doing this implies reencoding ucs2 files. The 2 buttons should also be disabled when you open the form if `ConfigValue "UTF-8 Encode All"` is True.

**Review by gilbertbw — CHANGES_REQUESTED** [review]

File: Modules/modFileEncoding.ACM (hunk 2 above)
Comment (OliverVahey) [reply Modules/modFileEncoding.ACM:526]: No, it's called in modExtendSaveAsText so it needs to be public

File: Modules/modExtendSaveAsText.ACM (hunk 1 above)
Comment (OliverVahey) [reply Modules/modExtendSaveAsText.ACM]: done

File: Forms/Access SVN - Options.ACF (hunk 3 above)
Comment (OliverVahey) [reply Forms/Access SVN - Options.ACF]: done

File: Forms/Access SVN - Options.ACF (hunk 4 above)
Comment (OliverVahey) [reply Forms/Access SVN - Options.ACF]: done

File: Modules/modExtendSaveAsText.ACM (hunk 5 above)
Comment (OliverVahey) [reply Modules/modExtendSaveAsText.ACM]: done

File: Forms/Access SVN - Options.ACF (hunk 7 above)
Comment (OliverVahey) [reply Forms/Access SVN - Options.ACF:2836]: done

File: Modules/modFileEncoding.ACM (hunk 2 above)
Comment (gilbertbw) [reply Modules/modFileEncoding.ACM:526]: OK

File: Forms/Access SVN - Options.ACF (hunk 7 above)
Comment (gilbertbw) [reply Forms/Access SVN - Options.ACF:2836]: You still need to do the second half of this

File: Modules/modExtendSaveAsText.ACM (hunk 6 above)
Comment (gilbertbw) [reply Modules/modExtendSaveAsText.ACM]: Done

**Comment by OliverVahey** [comment]
This work has been merged in, just via cherry-picking instead of PR

**Pull request closed without merging**